adr/adr04.md

```diff
@@ -3,10 +3,10 @@
 - **Title**: One REST API with multiple endpoints
 - **Status**: Accepted
 - **Context**:  
-  ClearView’s system spans multiple domains, including candidate management, employer management, job matching, and reporting. Initially, a separate REST API could be designed for each domain to create a clear separation of concerns. However, this introduces overhead in managing multiple services, increasing operational complexity and maintenance. To simplify the architecture and reduce the operational overhead, a decision has been made to consolidate all functionalities into a single REST API with multiple endpoints, where each endpoint handles a specific domain-related function (e.g., candidate, employer, matching, reporting).
+    ClearView’s system spans multiple domains, including candidate management, employer management, job matching, and reporting. To simplify the architecture and reduce operational overhead, a decision has been made to consolidate all functionalities into a unified API layer with multiple endpoints, where each endpoint handles a specific domain-related function (e.g., candidate, employer, matching, reporting).
```

Comment:
I would add sentence like this: This design would allow us to split services later if we need to scale per different endpoint. Having clear separation of components/concerns makes future changes much easier.